src/utils/regexp.ts

```diff
@@ -1,5 +1,33 @@
+import { Context } from "../api";
 import { assert } from "./errors";
 
+
+/**
+ * Wrapper around RegExp.exec that takes into account the user's smart case
+ * preference.
+ */
+export function smartExec(re: RegExp, string: string): RegExpExecArray | null {
+  const pattern = re.source;
+  const flags = re.flags;
+
+  if (!Context.current.extension.isSmartCaseEnabled) {
+    return re.exec(string);
+  }
+
+  const hasUpperCase = /[A-Z]/.test(pattern);
```

Comment:
I'd like to extend this to Unicode (in case a French person decides to lookup "Éditeur"):

```suggestion
  const hasUpperCase = /\p{Lu}/u.test(pattern);
```

Where `Lu` is the "uppercase letter" category. This will miss the case where the user uses `\p{Lu}` in their pattern, but I guess that's okay.